# Import Libs

In [1]:
import pandas as pd
import plotly.express as px

# Exploring Datasets

In [2]:
df_store = pd.read_csv("googleplaystore.csv")
df_reviews = pd.read_csv("googleplaystore_user_reviews.csv")

In [3]:
df_store.sample(10)

App       Category  Rating  Reviews  \
5294                   RULES OF SURVIVAL           GAME     4.2  1343106   
5787                    AX Battery Saver        FINANCE     4.6     4546   
6726       COMSATS BOOK STORE FOR BS(CS)         FAMILY     5.0       15   
8116        Cymath - Math Problem Solver         FAMILY     4.5    10159   
1174             HDFC Bank MobileBanking        FINANCE     4.2   208463   
4660                            U-Report         SOCIAL     4.3      241   
203         My Space - Employment Center       BUSINESS     4.5    67000   
1057                        Rabo Banking        FINANCE     3.4    31906   
10366        FG Radio - Radios de France         FAMILY     NaN        0   
993    Fandango Movies - Times + Tickets  ENTERTAINMENT     4.6   243747   

                     Size     Installs  Type Price Content Rating  \
5294                  56M  10,000,000+  Free     0           Teen   
5787                 3.0M     100,000+  Free     0       Everyone   
6726                  94M          50+  Free     0       Everyone   
8116                 6.4M   1,000,000+  Free     0       Everyone   
1174   Varies with device  10,000,000+  Free     0       Everyone   
4660                 7.5M      10,000+  Free     0           Teen   
203    Varies with device   1,000,000+  Free     0       Everyone   
1057   Varies with device   1,000,000+  Free     0       Everyone   
10366                6.5M          10+  Free     0       Everyone   
993    Varies with device  10,000,000+  Free     0           Teen   

              Genres      Last Updated         Current Ver         Android Ver  
5294          Action    August 1, 2018     1.180271.184729          4.0 and up  
5787         Finance  February 8, 2018               1.2.3          4.0 and up  
6726       Education     July 11, 2018                 2.0        4.0.3 and up  
8116       Education    April 13, 2018                2.17          4.1 and up  
1174         Finance     July 26, 2018  Varies with device  Varies with device  
4660          Social    April 10, 2018               2.8.4        4.0.3 and up  
203         Business     July 17, 2018  Varies with device          4.3 and up  
1057         Finance     July 19, 2018              5.16.0          4.0 and up  
10366  Entertainment     June 10, 2018                 1.0          4.2 and up  
993    Entertainment    August 2, 2018  Varies with device  Varies with device

In [4]:
df_store.drop(columns=["Current Ver", "Android Ver", "Last Updated"] , inplace=True)

In [5]:
df_reviews.head()

App                                  Translated_Review  \
0  10 Best Foods for You  I like eat delicious food. That's I'm cooking ...   
1  10 Best Foods for You    This help eating healthy exercise regular basis   
2  10 Best Foods for You                                                NaN   
3  10 Best Foods for You         Works great especially going grocery store   
4  10 Best Foods for You                                       Best idea us   

  Sentiment  Sentiment_Polarity  Sentiment_Subjectivity  
0  Positive                1.00                0.533333  
1  Positive                0.25                0.288462  
2       NaN                 NaN                     NaN  
3  Positive                0.40                0.875000  
4  Positive                1.00                0.300000

In [6]:
df_store.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
dtypes: float64(1), object(9)
memory usage: 847.1+ KB


In [7]:
df_store.describe()

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000

# Clean

In [8]:
df_store.dropna(inplace=True)

In [9]:
for col in df_store.columns:
    print(f"Unique in {col} col: ")
    print(df_store[col].unique())

Unique in App col: 
['Photo Editor & Candy Camera & Grid & ScrapBook' 'Coloring book moana'
 'U Launcher Lite – FREE Live Cool Themes, Hide Apps' ...
 'Fr. Mike Schmitz Audio Teachings' 'The SCP Foundation DB fr nn5n'
 'iHoroscope - 2018 Daily Horoscope & Astrology']
Unique in Category col: 
['ART_AND_DESIGN' 'AUTO_AND_VEHICLES' 'BEAUTY' 'BOOKS_AND_REFERENCE'
 'BUSINESS' 'COMICS' 'COMMUNICATION' 'DATING' 'EDUCATION' 'ENTERTAINMENT'
 'EVENTS' 'FINANCE' 'FOOD_AND_DRINK' 'HEALTH_AND_FITNESS' 'HOUSE_AND_HOME'
 'LIBRARIES_AND_DEMO' 'LIFESTYLE' 'GAME' 'FAMILY' 'MEDICAL' 'SOCIAL'
 'SHOPPING' 'PHOTOGRAPHY' 'SPORTS' 'TRAVEL_AND_LOCAL' 'TOOLS'
 'PERSONALIZATION' 'PRODUCTIVITY' 'PARENTING' 'WEATHER' 'VIDEO_PLAYERS'
 'NEWS_AND_MAGAZINES' 'MAPS_AND_NAVIGATION']
Unique in Rating col: 
[4.1 3.9 4.7 4.5 4.3 4.4 3.8 4.2 4.6 3.2 4.  4.8 4.9 3.6 3.7 3.3 3.4 3.5
 3.1 5.  2.6 3.  1.9 2.5 2.8 2.7 1.  2.9 2.3 2.2 1.7 2.  1.8 2.4 1.6 2.1
 1.4 1.5 1.2]
Unique in Reviews col: 
['159' '967' '87510' ... '603' '11

In [10]:
df_store.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9366 entries, 0 to 10840
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             9366 non-null   object 
 1   Category        9366 non-null   object 
 2   Rating          9366 non-null   float64
 3   Reviews         9366 non-null   object 
 4   Size            9366 non-null   object 
 5   Installs        9366 non-null   object 
 6   Type            9366 non-null   object 
 7   Price           9366 non-null   object 
 8   Content Rating  9366 non-null   object 
 9   Genres          9366 non-null   object 
dtypes: float64(1), object(9)
memory usage: 804.9+ KB


In [11]:
df_store.describe()

Rating
count  9366.000000
mean      4.191757
std       0.515219
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max       5.000000

In [12]:
df_store["Price"] = df_store["Price"].apply(lambda price : float(price.split("$")[-1]))

In [13]:
df_store["Price"].unique()

array([  0.  ,   4.99,   3.99,   6.99,   7.99,   5.99,   2.99,   3.49,
         1.99,   9.99,   7.49,   0.99,   9.  ,   5.49,  10.  ,  24.99,
        11.99,  79.99,  16.99,  14.99,  29.99,  12.99,   2.49,  10.99,
         1.5 ,  19.99,  15.99,  33.99,  39.99,   3.95,   4.49,   1.7 ,
         8.99,   1.49,   3.88, 399.99,  17.99, 400.  ,   3.02,   1.76,
         4.84,   4.77,   1.61,   2.5 ,   1.59,   6.49,   1.29, 299.99,
       379.99,  37.99,  18.99, 389.99,   8.49,   1.75,  14.  ,   2.  ,
         3.08,   2.59,  19.4 ,   3.9 ,   4.59,  15.46,   3.04,  13.99,
         4.29,   3.28,   4.6 ,   1.  ,   2.95,   2.9 ,   1.97,   2.56,
         1.2 ])

In [14]:
df_store.describe()

Rating        Price
count  9366.000000  9366.000000
mean      4.191757     0.960928
std       0.515219    15.816585
min       1.000000     0.000000
25%       4.000000     0.000000
50%       4.300000     0.000000
75%       4.500000     0.000000
max       5.000000   400.000000

# Analysis

# Q1: What is the top 3 highest category in term of ratings

In [22]:
df_groupby_category = df_store.groupby("Category").agg(["count" , "mean" , "min" , "max"])
df_groupby_category

Rating                     Reviews                 \
                     count      mean  min  max   count           mean   
Category                                                                
ART_AND_DESIGN          62  4.358065  3.2  5.0      62  2.580127e+218   
AUTO_AND_VEHICLES       73  4.190411  2.1  4.9      73  5.029587e+244   
BEAUTY                  42  4.278571  3.1  4.9      42  4.500119e+134   
BOOKS_AND_REFERENCE    178  4.346067  2.7  5.0     178            inf   
BUSINESS               303  4.121452  1.0  5.0     303            inf   
COMICS                  58  4.155172  2.8  5.0      58  1.747647e+222   
COMMUNICATION          328  4.158537  1.0  5.0     328            inf   
DATING                 195  3.970769  1.0  5.0     195            inf   
EDUCATION              155  4.389032  3.5  4.9     155            inf   
ENTERTAINMENT          149  4.126174  3.0  4.7     149            inf   
EVENTS                  45  4.435556  2.9  5.0      45  8.405336e+123   
FAMILY                1747  4.192272  1.0  5.0    1747            inf   
FINANCE                323  4.131889  1.0  5.0     323            inf   
FOOD_AND_DRINK         109  4.166972  1.7  5.0     109            inf   
GAME                  1097  4.286326  1.0  5.0    1097            inf   
HEALTH_AND_FITNESS     297  4.277104  1.4  5.0     297            inf   
HOUSE_AND_HOME          76  4.197368  2.8  4.8      76            inf   
LIBRARIES_AND_DEMO      65  4.178462  3.1  5.0      65  3.211666e+208   
LIFESTYLE              314  4.094904  1.5  5.0     314            inf   
MAPS_AND_NAVIGATION    124  4.051613  1.9  4.9     124            inf   
MEDICAL                350  4.189143  1.0  5.0     350            inf   
NEWS_AND_MAGAZINES     233  4.132189  1.7  5.0     233            inf   
PARENTING               50  4.300000  2.0  5.0      50  1.723588e+167   
PERSONALIZATION        314  4.335987  2.5  5.0     314            inf   
PHOTOGRAPHY            317  4.192114  2.0  5.0     317            inf   
PRODUCTIVITY           351  4.211396  1.0  5.0     351            inf   
SHOPPING               238  4.259664  1.6  5.0     238            inf   
SOCIAL                 259  4.255598  1.9  5.0     259            inf   
SPORTS                 319  4.223511  1.5  5.0     319            inf   
TOOLS                  734  4.047411  1.0  5.0     734            inf   
TRAVEL_AND_LOCAL       226  4.109292  2.2  5.0     226            inf   
VIDEO_PLAYERS          160  4.063750  1.8  4.9     160            inf   
WEATHER                 75  4.244000  3.3  4.8      75            inf   

                                     Price                         
                         min     max count      mean  min     max  
Category                                                           
ART_AND_DESIGN             1     967    62  0.096290  0.0    1.99  
AUTO_AND_VEHICLES      10479     994    73  0.027260  0.0    1.99  
BEAUTY                     1     964    42  0.000000  0.0    0.00  
BOOKS_AND_REFERENCE       10      98   178  0.134157  0.0    4.60  
BUSINESS                   1  984451   303  0.245512  0.0   17.99  
COMICS                  1006    9952    58  0.000000  0.0    0.00  
COMMUNICATION              1   99559   328  0.172835  0.0    4.99  
DATING                     1   97699   195  0.117744  0.0    7.99  
EDUCATION               1058   99020   155  0.115871  0.0    5.99  
ENTERTAINMENT          10291    9941   149  0.053557  0.0    4.99  
EVENTS                     1      99    45  0.000000  0.0    0.00  
FAMILY                     1     997  1747  1.314677  0.0  399.99  
FINANCE                    1     967   323  7.553777  0.0  399.99  
FOOD_AND_DRINK             1   91359   109  0.077798  0.0    4.99  
GAME                       1    9951  1097  0.255570  0.0   17.99  
HEALTH_AND_FITNESS         1     979   297  0.152795  0.0    7.99  
HOUSE_AND_HOME         10054     874    76  0.000000  0.0    0.00  
LIBRARIES_AND_DEMO     11087   

In [20]:
fig = px.bar(df_groupby_category["Rating"], x='mean', 
             y=df_groupby_category["Rating"].index,
             labels={"mean": "Ratings"},
                title="Category Vs Rating")
fig.show()

# Ans: 
## 1- Events with ~ 4.44<br>
## 2- Education with ~ 4.39<br>
## 3- Art And Design with ~ 4.36

# Q2: What is the top 3 category interaction (number of ratings)

In [21]:
fig = px.bar(df_groupby_category["Rating"], x='count', 
             y=df_groupby_category["Rating"].index,
             labels={"count": "Number of Ratings"},
                title="Category Vs Number of Rating")
fig.show()

# Ans: 
## 1- Family with 1747<br>
## 2- Game with 1097<br>
## 3- Tools with 734

# Discussion:
## As we can see there is a huge diffrence in numbers of rating from the previous top 3 with the current top 3, so the previous top 3 is not accurate perse, we could use Bayesian average instead to mitigate the difference in numbers but we will not dive into that

# Q3: What is the most popular genre in family category

In [26]:
df_family_category = df_store[df_store["Category"] == "FAMILY"]
df_family_category

App Category  Rating  Reviews  \
2014                       YouTube Kids   FAMILY     4.5   470694   
2015                         Candy Bomb   FAMILY     4.4    42145   
2016                             ROBLOX   FAMILY     4.5  4449910   
2017       Jewels Crush- Match 3 Puzzle   FAMILY     4.4    14774   
2018                   Coloring & Learn   FAMILY     4.4    12753   
...                                 ...      ...     ...      ...   
10820                   Fr. Daoud Lamei   FAMILY     5.0       22   
10827                Fr Agnel Ambarnath   FAMILY     4.2      117   
10834                     FR Calculator   FAMILY     4.0        7   
10836                  Sya9a Maroc - FR   FAMILY     4.5       38   
10837  Fr. Mike Schmitz Audio Teachings   FAMILY     5.0        4   

                     Size      Installs  Type  Price Content Rating  \
2014   Varies with device   50,000,000+  Free    0.0       Everyone   
2015                  20M   10,000,000+  Free    0.0       Everyone   
2016                  67M  100,000,000+  Free    0.0   Everyone 10+   
2017                  19M    1,000,000+  Free    0.0       Everyone   
2018                  51M    5,000,000+  Free    0.0       Everyone   
...                   ...           ...   ...    ...            ...   
10820                8.6M        1,000+  Free    0.0           Teen   
10827                 13M        5,000+  Free    0.0       Everyone   
10834                2.6M          500+  Free    0.0       Everyone   
10836                 53M        5,000+  Free    0.0       Everyone   
10837                3.6M          100+  Free    0.0       Everyone   

                             Genres  
2014    Entertainment;Music & Video  
2015             Casual;Brain Games  
2016   Adventure;Action & Adventure  
2017             Casual;Brain Games  
2018         Educational;Creativity  
...                             ...  
10820                     Education  
10827                     Education  
10834                     Education  
10836                     Education  
10837                     Education  

[1747 rows x 10 columns]

In [29]:
df_family_category_groupby = df_family_category.groupby("Genres").agg(["count" , "mean"])
df_family_category_groupby

Rating           Reviews                \
                                       count      mean   count          mean   
Genres                                                                         
Action;Action & Adventure                 14  4.300000      14  6.876090e+63   
Adventure;Action & Adventure               8  4.375000       8  5.562388e+45   
Adventure;Brain Games                      1  4.600000       1  7.148000e+03   
Adventure;Education                        2  4.100000       2  1.442616e+11   
Arcade;Action & Adventure                 15  4.346667      15  5.672343e+75   
...                                      ...       ...     ...           ...   
Strategy;Action & Adventure                2  4.600000       2  9.585102e+08   
Strategy;Creativity                        1  4.400000       1  1.295420e+05   
Strategy;Education                         1  4.500000       1  2.062000e+03   
Video Players & Editors;Creativity         1  4.100000       1  1.596190e+05   
Video Players & Editors;Music & Video      2  4.000000       2  1.988956e+10   

                                      Price            
                                      count      mean  
Genres                                                 
Action;Action & Adventure                14  0.640714  
Adventure;Action & Adventure              8  1.247500  
Adventure;Brain Games                     1  2.990000  
Adventure;Education                       2  0.000000  
Arcade;Action & Adventure                15  1.130667  
...                                     ...       ...  
Strategy;Action & Adventure               2  1.995000  
Strategy;Creativity                       1  0.000000  
Strategy;Education                        1  0.000000  
Video Players & Editors;Creativity        1  0.000000  
Video Players & Editors;Music & Video     2  0.000000  

[69 rows x 6 columns]

In [32]:
fig = px.bar(df_family_category_groupby["Rating"], x='mean', y=df_family_category_groupby.index,title="Genres Vs Rating")
fig.show()